src/ray/raylet/lib/python/common_extension.cc

```diff
@@ -111,19 +111,27 @@ int PyListStringToStringVector(PyObject *object,
 }
 
 static int PyObjectID_init(PyObjectID *self, PyObject *args, PyObject *kwds) {
-  const char *data;
+  const char *data = nullptr;
   int size;
-  if (!PyArg_ParseTuple(args, "s#", &data, &size)) {
+  if (!PyArg_ParseTuple(args, "|s#", &data, &size)) {
     return -1;
   }
-  if (size != sizeof(ObjectID)) {
+  if (data == nullptr) {
+    // Create the NIL object ID.
+    std::fill_n(self->object_id.mutable_data(), sizeof(ObjectID), 255);
+  } else if (size != sizeof(ObjectID)) {
     PyErr_SetString(CommonError, "ObjectID: object id string needs to have length 20");
```

Comment:
Here, I think it's more proper to raise a `ValueError` https://docs.python.org/2/library/exceptions.html#exceptions.ValueError